wpimath/src/main/java/edu/wpi/first/math/VecBuilder.java

```diff
@@ -27,7 +27,13 @@ public VecBuilder(Nat<N> rows) {
     super(rows, Nat.N1());
   }
 
-  private Vector<N> fillVec(double... data) {
+  /**
+   * Fills the vector with the given data.
+   *
+   * @param data The data to fill the vector with.
+   * @return The constructed vector.
+   */
+  public final Vector<N> fillVec(double... data) {
```

Comment:
Why was this made public?

Reply:
This was to allow a `VecBuilder` instance to be reused (instead of allocating a new instance for every call to `VecBuilder.fill(...)`) as well as for similarity between `MatBuilder` and `VecBuilder`. This could be changed back to private, though that would raise the question of if the constructor should be public, since then there's very marginal reason to use it.

Reply:
Yea, the constructor should probably be private. Furthermore, VecBuilder instances should be abolished in general since they don't offer any bookkeeping to the static fill() calls, and they just increase GC churn for commonly executed code.

The VecBuilder class could just contain static factory functions, but I'm not sure what's best to refactor the following MatBuilder code to reduce allocations:
```java
new MatBuilder<>(Nat.N2(), Nat.N1()).fill(leftVoltage, rightVoltage);
```

Reply:
It is a little weird for the row and column count to be right next to the data, but I think it's fine since they look so different from the data (different type and different construction)-You can tell them apart in `MatBuilder.fill(Nat.N2(), Nat.N1(), leftVoltage, rightVoltage)`. (Though in this case since there's exactly two doubles it is a bit more confusing than usual)

Reply:
Actually, this reminds me of this constructor overload: `public Matrix(Nat<R> rows, Nat<C> columns, double[] storage) {`.